beanie/odm/bulk.py

```diff
@@ -12,82 +14,140 @@
+_WriteOp = Union[
+    InsertOne[Mapping[str, Any]],
+    DeleteOne,
+    DeleteMany,
+    ReplaceOne[Mapping[str, Any]],
+    UpdateOne,
+    UpdateMany,
+]
 
 
 class BulkWriter:
+    """
+    A utility class for managing and executing bulk operations.
+
+    This class facilitates the efficient execution of multiple database operations
+    (e.g., inserts, updates, deletes, replacements) in a single batch. It supports asynchronous
+    context management and ensures that all queued operations are committed upon exiting the context.
+
+    Attributes:
+        session Optional[AsyncIOMotorClientSession]:
+            The motor session used for transactional operations.
+            Defaults to None, meaning no session is used.
+        ordered Optional[bool]:
+            Specifies whether operations are executed sequentially (default) or in parallel.
+            - If True, operations are performed serially, stopping at the first failure.
+            - If False, operations may be executed in arbitrary order, and all operations are attempted
+              regardless of individual failures.
+        bypass_document_validation Optional[bool]:
+            If True, document-level validation is bypassed for all operations
+            in the bulk write. This applies to MongoDB's schema validation rules, allowing documents that
+            do not meet validation criteria to be inserted or modified. Defaults to False.
+        comment Optional[Any]:
+            A user-provided comment attached to the bulk operation command, useful for
+            auditing and debugging purposes.
+        operations List[Union[DeleteMany, DeleteOne, InsertOne, ReplaceOne, UpdateMany, UpdateOne]]:
+            A list of MongoDB operations queued for bulk execution.
+
+    Parameters:
+        session Optional[AsyncIOMotorClientSession]: The motor session for transaction support.
+            Defaults to None (no session).
+        ordered Optional[bool]: Specifies whether operations are executed in sequence (True) or in parallel (False).
+            Defaults to True.
+        bypass_document_validation Optional[bool]: Allows the bulk operation to bypass document-level validation.
+            This is particularly useful when working with schemas that are being phased in or for bulk imports
+            where strict validation may not be necessary. Defaults to False.
+        comment Optional[Any]: A custom comment attached to the bulk operation.
+            Defaults to None.
+    """
+
     def __init__(
         self,
         session: Optional[AsyncIOMotorClientSession] = None,
         ordered: bool = True,
-    ):
-        self.operations: List[Operation] = []
+        bypass_document_validation: bool = False,
+        comment: Optional[Any] = None,
+    ) -> None:
+        self.operations: List[_WriteOp] = []
         self.session = session
         self.ordered = ordered
+        self.object_class: Optional[Type[Union[Document, UnionDoc]]] = None
```

Comment:
I think you can add object class as a constructor parameter here. It should not break any logic but will allow set up the correct class on the very first step